src/vorta/views/diff_result.py

```diff
@@ -219,3 +233,49 @@ def flags(self, index):
         if not index.isValid():
             return Qt.NoItemFlags
         return Qt.ItemIsEnabled
+
+
+class SortProxyModel(QSortFilterProxyModel):
+    def __init__(self, model):
+        super().__init__()
+        self.setSourceModel(model)
+
+    def lessThan(self, left: QModelIndex, right: QModelIndex):
+        left_data = left.data(Qt.UserRole)
+        right_data = right.data(Qt.UserRole)
+        sort_order = self.sortOrder()
+        sort_column = self.sortColumn()
+        left_total_bytes = 0
+        right_total_bytes = 0
+
+        if sort_column == 0:
+            left_is_folder = not "bytes" in left_data
+            right_is_folder = not "bytes" in right_data
+            left_value = left_data["name"].upper()
+            right_value = right_data["name"].upper()
+
+            # Keep folders at the top and continue to sort by name
+            if left_is_folder and not right_is_folder:
+                if sort_order == Qt.SortOrder.AscendingOrder:
+                    left_value = "!" + left_value
+                else:
+                    right_value = "!" + right_value
+            if right_is_folder and not left_is_folder:
+                if sort_order == Qt.SortOrder.AscendingOrder:
+                    right_value = "!" + right_value
+                else:
+                    left_value = "!" + left_value
+        elif sort_column == 1:
+            left_value = left_data["modified"]
+            right_value = right_data["modified"]
+        elif sort_column == 2:
+            try:
+                left_value = left_data["bytes"]
+            except KeyError:
+                left_value = left_total_bytes
+            try:
+                right_value = right_data["bytes"]
+            except KeyError:
+                right_value = right_total_bytes
+
+        return left_value < right_value
```

Comment:
This seems way to complicated for implementing a *lower than*. I especially think that the use of the sort order and sort column attribute might even be wrong inside this method. The use of the `UserRole` also seems unnecessary since the internal pointer of the indices will reference the corresponding `FolderItem`. 

This is how simple I did it:
https://github.com/real-yfprojects/vorta/blob/sortproxymodel/src/vorta/views/diff_result.py#L240-L262